instrumentation/spring/spring-boot-actuator-autoconfigure-2.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/actuator/v2_0/ActuatorTest.java

```diff
@@ -57,9 +60,19 @@ void shouldInjectOtelMeterRegistry() {
                                                         "value"))))));
 
     MeterRegistry meterRegistry = context.getBean(MeterRegistry.class);
-    assertThat(meterRegistry).isNotNull().isInstanceOf(CompositeMeterRegistry.class);
-    assertThat(((CompositeMeterRegistry) meterRegistry).getRegistries())
-        .anyMatch(r -> r.getClass().getSimpleName().equals("OpenTelemetryMeterRegistry"))
-        .anyMatch(r -> r.getClass().getSimpleName().equals("SimpleMeterRegistry"));
+    assertThat(meterRegistry).isInstanceOf(CompositeMeterRegistry.class);
+    AbstractCollectionAssert<
+            ?, Collection<? extends MeterRegistry>, MeterRegistry, ObjectAssert<MeterRegistry>>
+        match =
+            assertThat(((CompositeMeterRegistry) meterRegistry).getRegistries())
+                .anyMatch(r -> r.getClass().getSimpleName().equals("OpenTelemetryMeterRegistry"))
+                .anyMatch(r -> r.getClass().getSimpleName().equals("SimpleMeterRegistry"));
+
+    try {
+      Class.forName("io.micrometer.prometheusmetrics.PrometheusMeterRegistry");
+      match.anyMatch(r -> r.getClass().getSimpleName().equals("PrometheusMeterRegistry"));
+    } catch (ClassNotFoundException e) {
+      // not testing prometheus
+    }
```

Comment:
when running this test the prometheus dependency is never available as far as I can tell. Consider whether you want to test the prometheus here at all. If you want then instead of adding a suite consider adding the prometheus dependency to the main test dependencies.

Reply:
after the fix (https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/13083#discussion_r1946434841) it's failing on this line - after the prometheus class is loaded

Reply:
`io.micrometer.prometheus.PrometheusMeterRegistry` that is missing and `io.micrometer.prometheusmetrics.PrometheusMeterRegistry` are different classes. The former is deprecated and comes from a different jar.

Reply:
great catch!

I think we can only run the test in latest dep - where it works